pkg/ingress/model_build_load_balancer.go

```diff
@@ -309,18 +310,27 @@ func (t *defaultModelBuildTask) buildLoadBalancerSecurityGroups(ctx context.Cont
 			if !t.enableBackendSG {
 				return nil, errors.New("backendSG feature is required to manage worker node SG rules when frontendSG manually specified")
 			}
-			backendSGID, err := t.backendSGProvider.Get(ctx)
+			backendSGID, err := t.backendSGProvider.Get(ctx, t.getActiveIngressObjects(ctx))
 			if err != nil {
 				return nil, err
 			}
 			t.backendSGIDToken = core.LiteralStringToken(backendSGID)
+			t.backendSGAllocated = true
 			lbSGTokens = append(lbSGTokens, t.backendSGIDToken)
 		}
 		t.logger.Info("SG configured via annotation", "LB SGs", lbSGTokens, "backend SG", t.backendSGIDToken)
 	}
 	return lbSGTokens, nil
 }
 
+func (t *defaultModelBuildTask) getActiveIngressObjects(_ context.Context) []metav1.Object {
+	var activeIngresses []metav1.Object
```

Comment:
see my previous comment about getActiveAndInactiveIngressObjects, this can be a utility function

Reply:
updated